twake/frontend/src/app/components/ChatUploads/UploadsViewer.tsx

```diff
@@ -0,0 +1,27 @@
+import React, { useEffect } from 'react';
+import { PendingFilesListState } from 'app/state/recoil/atoms/PendingFilesList';
+import { useRecoilState } from 'recoil';
+import ChatUploadServiceManager, { ChatUploadService } from './ChatUploadService';
+
+import PendingFilesList from 'app/components/FileComponents/PendingFilesList';
+
+let chatUploadService: ChatUploadService;
+const ChatUploadsViewer = (): JSX.Element => {
+  const [pendingFilesListState, setPendingFilesListState] = useRecoilState(PendingFilesListState);
+
+  useEffect(() => {
+    chatUploadService = ChatUploadServiceManager.get();
+    chatUploadService.setHandler(setPendingFilesListState);
```

Comment:
I think that what we discussed together is this. We may not need to access the ChatUploadServiceManager instance like this, instead we should have a custom hook like `useUploadService` which returns what you need and which can be reused across all the components.

In the hook, you will have access to all the recoil hooks and so you do not need the `setHandler`, and you will not need the `notify` function in it, just relying on recoil will be OK.

Reply:
Yes, it was the initial idea, I don't know why we kept, this, I call Stephane to do the small update

Reply:
BTW, using `useRef` to get the singleton of the service if needed will be the way to go.

Reply:
We need to discuss again about this. Let's plan 15 minutes tomorrow after daily

Reply:
I just updated code, should i close this ? @RomaricMourgues